waku/waku_core/topics/pubsub_topic.nim

```diff
@@ -89,49 +72,23 @@ proc parseStaticSharding*(
 
   ok(NsPubsubTopic.staticSharding(clusterId, shardId))
 
-proc parseNamedSharding*(
-    T: type NsPubsubTopic, topic: PubsubTopic | string
-): ParsingResult[NsPubsubTopic] =
-  if not topic.startsWith(Waku2PubsubTopicPrefix):
-    return err(ParsingError.invalidFormat("must start with " & Waku2PubsubTopicPrefix))
-
-  let raw = topic[8 ..< topic.len]
-  if raw.len == 0:
-    return err(ParsingError.missingPart("topic-name"))
-
-  ok(NsPubsubTopic.named(name = raw))
-
 proc parse*(
     T: type NsPubsubTopic, topic: PubsubTopic | string
 ): ParsingResult[NsPubsubTopic] =
   ## Splits a namespaced topic string into its constituent parts.
   ## The topic string has to be in the format `/<application>/<version>/<topic-name>/<encoding>`
-  if topic.startsWith(StaticShardingPubsubTopicPrefix):
-    NsPubsubTopic.parseStaticSharding(topic)
-  else:
-    NsPubsubTopic.parseNamedSharding(topic)
+  NsPubsubTopic.parseStaticSharding(topic)
```

Comment:
In this `proc` I think we need to change the `topic` type to only `PubsubTopic`

Reply:
I see that
https://github.com/waku-org/nwaku/blob/8b42f199baf4e00794c4cec4d8601c3f6c330a20/waku/waku_core/topics/pubsub_topic.nim#L17

So will try to change the places where it says
```
PubsubTopic | string
```

to just `PubsubTopic`, if that makes sense